### PR TITLE
Create GitHub templates

### PR DESCRIPTION
**Pull request by ravernkoh**
Adds a template for Issue and Pull Request descriptions, and also a Contributing guide.

### Remarks

* These guides should suffice for now
* In the future when more people work on this project, they should be updated with more information and description

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Summary
+
+_Give a brief description of the problem._
+
+## Remarks
```
Comment (sean1188) [inline .github/ISSUE_TEMPLATE.md]: This should be a h3 header for aesthetics.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Summary
+
+_Give a brief description of what was added/changed/removed._
+
+## Remarks
```
Comment (sean1188) [inline .github/PULL_REQUEST_TEMPLATE.md]: This should be a h3 header for aesthetics.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## Commit messages
+
+Commit messages start with a captial letter. The first word should be a verb to represent what the main intention is. For example, a commit intended to fix a bug would have a message that goes something like `Fix major bug in core`. Other verbs include `Create`, `Add`, `Refactor`, `Change`, `Remove` and `Delete`.
+
+## Branching
+
+Branch names start with verbs to represent what the main intention is. For example, a branch intended to add a new feature would be `create-new-feature`. Other verbs include `add`, `refactor`, `change`, `fix`, `remove` and `delete`.
+
+Additionally, if the branch only concerns a single package, the branch name should be preceded with the package that the branch is intended for. For example, branches that work on the `api` package should be named `api/branch-name`. For branches that concern the repository in general, the branch name should remain as is.
```
Comment (yadunut) [inline .github/CONTRIBUTING.md]: For branching I prefer we follow [Git Flow ](https://gitversion.readthedocs.io/en/latest/git-branching-strategies/gitflow-examples/). So hotfixes would be under `hotfix/<hotfix-name>`, and features would be under `feature/<feature-name>`. We can still follow the naming commit naming convention as above.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (ravernkoh) [reply .github/ISSUE_TEMPLATE.md]: 👍

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (ravernkoh) [reply .github/CONTRIBUTING.md]: What is your opinion on the branches for packages? Should it be `api/feature/create-auth-module` or `feature/api/create-auth-module`? Should we even segment it by package?

Also, will it be `feature/auth-module` or `feature/create-auth-module`?

Comment (sean1188) [reply .github/CONTRIBUTING.md]: I think `feature/package/auth-module` makes most sense.

Comment (ravernkoh) [reply .github/CONTRIBUTING.md]: How is it done in your company @yadunut?

**Comment by ravernkoh** [comment]
Removed the summary heading. Felt that it is redundant.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (yadunut) [reply .github/CONTRIBUTING.md]: I agree with @sean1188. We use both `feature/api/create-auth-module` and `feature/api/auth-module`. I feel that we should let naming of adding such  details as `create` go to the branch owner. When we are merging the PR to the main branch, the branch name need not be in the commit message header.

**Comment by ravernkoh** [comment]
Closing this PR and creating a new branch with the proper naming.

**Pull request closed without merging**